### PR TITLE
add grpc api of container signal

### PR DESCRIPTION
**Pull request by Crazykev**
Signed-off-by: haozhang crazykev@zju.edu.cn

### CODE REVIEW

File: integration/client.go [hunk 1]
```diff
@@ -614,3 +614,17 @@ func (c *HyperClient) Info() (*types.InfoResponse, error) {
 
 	return info, nil
 }
+
+// ContainerSignal sends a signal to specified container of specified pod
+func (c *HyperClient) ContainerSignal(podID, containerID string, signal int64) error {
```
Comment (laijs) [inline integration/client.go]: where are the test code to call it?

**Comment by Crazykev** [comment]
@laijs added test case

**Comment by Crazykev** [comment]
@laijs updated and rebased.

File: types/types.proto [hunk 2]
```diff
@@ -759,6 +759,14 @@ message PingResponse {
   string hyperdStats = 1;
 }
 
+message ContainerSignalRequest {
+    string podID     = 1;
+    string containerID = 2;
+    int64  signal    = 3;
```
Comment (laijs) [inline types/types.proto]: could you please fix the indent?

Comment (Crazykev) [inline types/types.proto]: Sure, rely too much on gofmt, fixed that.

**Comment by laijs** [comment]
need rebase. see #482

**Comment by gnawux** [comment]
woo... Sorry for introduce conflict for you...

**Comment by Crazykev** [comment]
Rebased.
@gnawux either of us need to do that :wink:

**Comment by laijs** [comment]
```
I1221 05:21:54.559450   22372 server.go:152] Calling GET /v0.7.0/images/get
# github.com/hyperhq/hyperd/integration
/home/travis/gopath/src/github.com/hyperhq/hyperd/integration/hyper_test.go:517: too many arguments in call to s.client.StartPod
FAIL	github.com/hyperhq/hyperd/integration [build failed]
!!! Error in /home/travis/gopath/src/github.com/hyperhq/hyperd/hack/lib/test.sh:129
  'go test github.com/hyperhq/hyperd/integration -check.vv -v' exited with status 2
Call stack:
  1: /home/travis/gopath/src/github.com/hyperhq/hyperd/hack/lib/test.sh:129 hyper::test::integration(...)
  2: hack/test-cmd.sh:161 runTests(...)
  3: hack/test-cmd.sh:191 main(...)
Exiting with status 1
I1221 05:21:59.463920   22372 daemon.go:345] The daemon will be shutdown
I1221 05:21:59.464034   22372 daemon.go:346] Shutdown all VMs
+++ [1221 05:21:59] Clean up complete
```

**Comment by Crazykev** [comment]
@laijs  fixed.  Seems failure in CI not concerned with this patch.
>!!! Error in /home/travis/gopath/src/github.com/hyperhq/hyperd/hack/lib/test.sh:51
  'sudo env PATH=$PATH hyperctl run --rm -a -p $1' exited with status 1
Call stack:
  1: /home/travis/gopath/src/github.com/hyperhq/hyperd/hack/lib/test.sh:51 hyper::test::run_attached_pod(...)
  2: /home/travis/gopath/src/github.com/hyperhq/hyperd/hack/lib/test.sh:67 hyper::test::insert_file(...)
  3: hack/test-cmd.sh:171 runTests(...)
  4: hack/test-cmd.sh:191 main(...)
Exiting with status 1

**Comment by laijs** [comment]
close it, and reopen it later with the hope that it will be tested with the newest hyperd

**Comment by Crazykev** [comment]
@laijs travis can pass now.

**Comment by gnawux** [comment]
retest this please, @hykins

**Comment by laijs** [comment]
could you also add the signal api for process(exec) please?

**Comment by Crazykev** [comment]
@laijs yep, will do

**Pull request merged**